sdlib/d/gc/slab.d

```diff
@@ -214,14 +263,13 @@ private:
 			return;
 		}
 
-		writePackedFreeSpace(freeSpacePtr, size);
 		if (!_hasMetadata) {
 			e.enableMetadata(index);
 			_hasMetadata = true;
 		}
-	}
 
-	enum FinalizerBit = nativeToBigEndian!size_t(0x2);
+		slotMetadata.setFreeSpace(size);
```

Comment:
I don't get why you reorder these things. Is there a reason?

Reply:
Yes, see above. Calling `slotMetadata` without `_hasMetadata` being true would be an assert.

I made it a public function so I could use it outside, and I figured that is a sensible check. Do you think the reordering will have some effect?

Reply:
Oh! I realized, I can make slotMetadata a private accessor, and then I don't have to worry about external users calling on non-metadata items. I'll fix this.